backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/geocoding/query.kt

```diff
@@ -0,0 +1,72 @@
+package app.ehrenamtskarte.backend.stores.geocoding
+
+import app.ehrenamtskarte.backend.common.COUNTRY_CODE
+import app.ehrenamtskarte.backend.common.STATE
+import app.ehrenamtskarte.backend.stores.importer.types.AcceptingStore
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.ktor.client.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import org.geojson.Feature
+import org.geojson.FeatureCollection
+import java.io.File
+
+suspend fun queryFeatures(params: List<Pair<String, String>>): List<Feature> {
+    val fileName = cacheFileName(params)
+    val file = File("${System.getProperty("user.dir")}/data/nominatim/v1/$fileName.json")
+
+    val geoJson = if (file.exists()) {
+        // Use a cached version on disk.
+        file.readText()
+    } else {
+        val client = HttpClient()
+        val response = client.request<String> {
+            url {
+                protocol = URLProtocol.HTTP
+                host = System.getProperty("app.geocoding.host")
+                path("nominatim", "search")
+                parameters.append("format", "geojson")
+                parameters.append("addressdetails", "1")
+                parameters.append("countrycodes", COUNTRY_CODE)
+                params.forEach {
+                    parameters.append(it.first, it.second)
+                }
+            }
+            method = HttpMethod.Get
+        }
+        file.parentFile.mkdirs()
+        file.writeText(response)
```

Comment:
We need to handle here a few cases if we do the caching ourselves:

* Maximum cache size (max. files and max. size in bytes)
* Invalidation
* Handling errors like: Out of Storage, No permission etc.

Reply:
Can we move the caching topic to a separate PR/Issue? If you want, I can also remove it completely for now.

Reply:
https://github.com/digitalfabrik/ehrenamtskarte/issues/445

Reply:
Will remove the caching for now